### PR TITLE
Create YaoTreeToPastaQList.jl

### PR DESCRIPTION
**Pull request by VarLad**
W.I.P
Incomplete, as of yet

### CODE REVIEW

**Comment by Roger-luo** [comment]
we use snake case for file names except the module file. for this package, it should be simple enough to just put code inside `src/YaoPastaQ.jl`

**Comment by Roger-luo** [comment]
we use snake case for file names except the module file. for this package, it should be simple enough to just put code inside `src/YaoPastaQ.jl`

**Comment by VarLad** [comment]
@Roger-luo thats most of the gates
Can't find the equivalent for 
```iSWAP, Rn, sqrt(X), CRn, sqrt(SWAP) gates are left```

File: Manifest.toml [hunk 1]
```diff
@@ -0,0 +1,308 @@
+# This file is machine-generated - editing it directly is not advised
+
+[[ArgTools]]
```
Comment (Roger-luo) [inline Manifest.toml:3]: can can put Manifest.toml in `.gitignore`?

File: Project.toml [hunk 2]
```diff
@@ -3,6 +3,10 @@ uuid = "295aef9f-e68b-4375-90fc-eaec2cf620b6"
 authors = ["Roger-Luo <rogerluo.rl18@gmail.com> and contributors"]
 version = "0.1.0"
 
+[deps]
+Yao = "5872b779-8223-5990-8dd0-5abbb0748c8c"
```
Comment (Roger-luo) [inline Project.toml:7]: you don't need to depend on both, Yao is a meta-package, you should just depend on YaoBlocks and add YaoBase as dependency to get register API

File: src/YaoPastaQ.jl [hunk 3]
```diff
@@ -1,5 +1,56 @@
 module YaoPastaQ
+	
+	using Yao, YaoBlocks
```
Comment (Roger-luo) [inline src/YaoPastaQ.jl:3]: please follow the format guide https://github.com/QuantumBFS/Yao.jl/blob/master/CONTRIBUTING.md#core-formatting-guidelines

File: src/YaoPastaQ.jl [hunk 4]
```diff
@@ -1,5 +1,56 @@
...
+	    genlist!(plist, blk.content, sublocs(blk.locs, locs), controls)
+	end
+
+	genlist!(plist, blk::XGate, locs, controls) = push!(plist, ("X", locs[1]))
+	genlist!(plist, blk::YGate, locs, controls) = push!(plist, ("Y", locs[1]))
+	genlist!(plist, blk::HGate, locs, controls) = push!(plist, ("H", locs[1]))
+	genlist!(plist, blk::ZGate, locs, controls) = push!(plist, ("Z", locs[1]))
+	genlist!(plist, blk::Scale{Val{im}, 1, YGate}, locs, controls) = push!(plist, ("iY", locs[1]))
+	genlist!(plist, blk::TGate, locs, controls) = push!(plist, ("π/8", locs[1]))
+	genlist!(plist, blk::ShiftGate{Float64}, locs, controls) =   if(rad2deg(blk.theta) == 90.0)
```
Comment (Roger-luo) [inline src/YaoPastaQ.jl:36]: use `function genlist!` here instead of `=`

Comment (Roger-luo) [inline src/YaoPastaQ.jl:36]: this is wrong

```suggestion
	genlist!(plist, blk::ShiftGate{Float64}, locs, controls) =   if rad2deg(blk.theta) == 90.0
```

Comment (Roger-luo) [reply src/YaoPastaQ.jl:36]: please remove all `rad2deg` function and just use radius for comparision.

**Review by Roger-luo — CHANGES_REQUESTED** [review]
this is still far from finish, you still need:

1. make a PastaQ regsiter
2. write tests to get over 90% test coverage

**Comment by VarLad** [comment]
@Roger-luo Okay, kindly review it once more. I'll start writing the tests if its okay this far

**Comment by Roger-luo** [comment]
@VarLad you didn't implement the register type.

please do a `git rm Manifest.toml` to remove Manifest.toml file from this PR

you are not following the style guide, code inside module shouldn't have indent

**Comment by Roger-luo** [comment]
> I'll start writing the tests if its okay this far

please add tests first before asking for review

**Comment by VarLad** [comment]
> you are not following the style guide, code inside module shouldn't have indent

Sorry, I saw this :sweat_smile: 
```
General Formatting Guidelines for Julia code contributions

    4 spaces per indentation level, no tabs
```
https://github.com/JuliaLang/julia/blob/master/CONTRIBUTING.md#general-formatting-guidelines-for-julia-code-contributions

I'll remove the indentation

**Comment by VarLad** [comment]
@Roger-luo If this is all correct, I guess what remains is to write the register.....

File: test/runtests.jl [hunk 5]
```diff
@@ -1,6 +1,14 @@
-using YaoPastaQ
+using YaoPastaQ, Yao
```
Comment (Roger-luo) [inline test/runtests.jl:1]: move Yao to test dependency

File: Project.toml [hunk 6]
```diff
@@ -3,6 +3,11 @@ uuid = "295aef9f-e68b-4375-90fc-eaec2cf620b6"
 authors = ["Roger-Luo <rogerluo.rl18@gmail.com> and contributors"]
```
Comment (Roger-luo) [inline Project.toml:3]: add yourself

File: test/runtests.jl (hunk 5 above)
Comment (VarLad) [reply test/runtests.jl:1]: Uhhh... in Project.toml?

**Comment by Roger-luo** [comment]
looks good in general, we can merge this PR after change the above two coments. then you can open a new PR for the register

**Comment by VarLad** [comment]
Okay, now the registers....

**Pull request merged**